accounting/categories.go

```diff
@@ -14,6 +16,25 @@ type CustomCategory struct {
 	Regexes []*regexp.Regexp
 }
 
+// NewCustomCategory creates compiles the set of regexes provided and returning
+// a new category. This function assumes that each regex is unique.
+func NewCustomCategory(name string, regexes []string) (*CustomCategory, error) {
```

Comment:
Is there a specific reason why a pointer is returned here while the slice always contains non-pointer elements? IMO we should always use pointers to structs. Or is the main goal to avoid nil references?

Reply:
I used a pointer here mostly because it's convenient to be able to return nil values on errors, and as you say it generally makes sense to use pointers for structs.

As for the slice, I'm trying to get out of the habit of slices of pointers generally: https://philpearl.github.io/post/bad_go_slice_of_pointers/. I think that we discussed this a while back as a team, or in some channel, but never really acted on it. I think that it makes sense to not use slices of pointers as a rule, but it does have a bit of friction with pointers-everywhere-for-structs, because it seems a bit unintuitive as you point out.

Reply:
Thanks for the article, that makes a lot of sense. I guess the reason why I find slices of structs weird is because you cannot tell the difference between a member that happens to use all default values vs. a member that wasn't initialized at all.
With pointers you would at least run into a nil reference error instead of just silently using the default values. But that probably almost never leads to issues if you're just looping over the whole slice anyway.

Reply:
Yeah, I go back and forth on it myself. Perhaps worth a more general discussion re what we want to do across projects.